engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/TreeTableImpl.java

```diff
@@ -222,6 +222,28 @@ protected TreeTableImpl copy() {
                 parentIdentifierColumn, nodeFilterColumns, nodeOperations, availableColumnDefinitions);
     }
 
+    @Override
+    public TreeTable rebase(@NotNull final Table newSource) {
+        if (!newSource.getDefinition().equals(source.getDefinition())) {
```

Comment:
We could make a case that we just need to have the parent and id columns present on `newSource`. I guess we don't need to do that for the ACL case.